acme/client.go

```diff
@@ -167,15 +167,29 @@ func (c *Client) Register() (*RegistrationResource, error) {
 	}
 
 	var serverReg Registration
+	var regURI string
 	hdr, err := postJSON(c.jws, c.directory.NewRegURL, regMsg, &serverReg)
 	if err != nil {
-		return nil, err
+		remoteErr, ok := err.(RemoteError)
+		if ok && remoteErr.StatusCode == 409 {
+			regURI = hdr.Get("Location")
+			regMsg.Resource = "reg"
+			if hdr, err = postJSON(c.jws, regURI, regMsg, &serverReg); err != nil {
```

Comment:
Nope, at least to my knowledge this is just to recover existing registration information as stated in the [ACME protocol specification](https://letsencrypt.github.io/acme-spec/#registration). 
In case you use a new account key a new registration should be created and we don't get a 409 back.